tools/operator/connectionprofile/connectionprofile.go

```diff
@@ -91,6 +100,221 @@ func (c ConnProfile) GenerateConnProfilePerOrg(orgName string) error {
+
+	return caliperConnProfile, nil
+}
+
+func (c ConnProfile) clientCaliper(orgName string) (map[string]networkspec.CaliperClient, error) {
+	clients := make(map[string]networkspec.CaliperClient)
+	var client networkspec.CaliperClient
+
+	client.Client.Organization = orgName
+	client.Client.CredentialStore.Path = fmt.Sprintf("/tmp/%s", orgName)
+	client.Client.CredentialStore.CryptoStore.Path = fmt.Sprintf("/tmp/%s", orgName)
+	client.Client.Conenction.Timeout.Peer.Endorser = 300
+	client.Client.Conenction.Timeout.Peer.EventHub = 600
+	client.Client.Conenction.Timeout.Peer.EventReg = 300
+	client.Client.Conenction.Timeout.Orderer = 300
+	peerOrgsPath := paths.PeerOrgsDir(c.Config.ArtifactsLocation)
+
+	//privateKey
+	keystorePath := paths.JoinPath(peerOrgsPath, fmt.Sprintf("%s/users/User1@%s/msp/keystore", orgName, orgName))
+	privKeyFile, err := ioutil.ReadDir(keystorePath)
+	if err != nil {
+		return clients, err
+	}
+	privKeyPath := paths.JoinPath(keystorePath, fmt.Sprintf("%s", privKeyFile[0].Name()))
+	cert, err := c.GetCertificateFromFile(privKeyPath)
+	if err != nil {
+		return clients, err
+	}
+	client.Client.ClientPrivateKey.Pem = cert
+
+	//signedCerts
+	signCertsPath := paths.JoinPath(peerOrgsPath, fmt.Sprintf("%s/users/User1@%s/msp/signcerts", orgName, orgName))
+	signCertsFile, err := ioutil.ReadDir(signCertsPath)
+	if err != nil {
+		return clients, err
+	}
+	userCertsPath := paths.JoinPath(signCertsPath, fmt.Sprintf("%s", signCertsFile[0].Name()))
+	cert, err = c.GetCertificateFromFile(userCertsPath)
+	if err != nil {
+		return clients, err
+	}
+	client.Client.ClientSignedCert.Pem = cert
+	clientName := fmt.Sprintf("client-%s", orgName)
+	clients[clientName] = client
+
+	return clients, nil
+}
+
+func (c ConnProfile) channelCaliper(orgName string, cprofile CaliperConnProfile) (map[string]networkspec.CaliperChannel, error) {
+	channels := make(map[string]networkspec.CaliperChannel)
+
+	// for each channel
+	for i := 0; i < c.Config.NumChannels; i++ {
+		var channel networkspec.CaliperChannel
+		//channel created
+		channel.Created = true
+
+		//orderers
+		ordererMap := cprofile.Orderers
+		for o := range ordererMap {
+			channel.Orderers = append(channel.Orderers, o)
+		}
+
+		//peers
+		peerMap := cprofile.Peers
+		pr := make(map[string]networkspec.CaliperChannelPeer)
+		for pName := range peerMap {
+			var p networkspec.CaliperChannelPeer
+			p.EventSource = true
+			pr[pName] = p
+		}
+		channel.Peers = pr
+
+		//chaincode
+		// for chaincodes loop
+		for j := 0; j < len(c.Config.ChaincodeIDs); j++ {
+			var cc networkspec.ChaincodeID
+			cc.Id = c.Config.ChaincodeIDs[j].Id
+			cc.Version = c.Config.ChaincodeIDs[j].Version
+			channel.Chaincodes = append(channel.Chaincodes, cc)
+		} // end chaincodeIDs loop
+
+		chName := c.Config.ChannelPrefix
+		if len(chName) == 0 {
+			chName = "testorgschannel"
+		}
+		chName = fmt.Sprintf("%s%d", chName, i)
+		channels[chName] = channel
+	} // end channel loop
+
+	return channels, nil
+}
```

Comment:
In general its not idiomatic to assign object values the same way we do in nodejs, we initialize structs rather than indexing into them like `a.b.c.d`. This function can be rewritten as:

```
channels := make(map[string]networkspec.CaliperChannel)
	for i := 0; i < c.Config.NumChannels; i++ {
		orderers := make([]string, len(cprofile.Orderers))
		for orderer := range cprofile.Orderers {
			orderers = append(orderers, orderer)
		}

		peers := make(map[string]networkspec.CaliperChannelPeer, len(cprofile.Peers))
		for peer := range cprofile.Peers {
			peers[peer] = networkspec.CaliperChannelPeer{EventSource: true}
		}

		chaincodes := make([]networkspec.ChaincodeID, len(c.Config.ChaincodeIDs))
		for _, chaincode := range c.Config.ChaincodeIDs {
			cc := networkspec.ChaincodeID{
				Id:      chaincode.Id,
				Version: chaincode.Version,
			}
			chaincodes = append(chaincodes, cc)
		}

		channel := networkspec.CaliperChannel{
			Created:    true,
			Chaincodes: chaincodes,
			Orderers:   orderers,
			Peers:      peers,
		}

		chName := c.Config.ChannelPrefix
		if chName == "" {
			chName = "testorgschannel"
		}
		chName = fmt.Sprintf("%s%d", chName, i)
		channels[chName] = channel
	}
	return channels
}
```

Reply:
ok